kbase-extension/static/kbase/js/widgets/common/errorMessage.js

```diff
@@ -0,0 +1,165 @@
+define(['jquery', 'jsonrpc/1.1/errors'], ($, errors) => {
+    'use strict';
+
+    /**
+     *
+     *  message - canonical message
+     *     error - may be an object with ...
+     *        name - string - always JSONRPCError
+     *        code - number - JSONRPC compatible error code
+     *        message - string - canonical message
+     *        error - string - captures trace
+     *
+     */
+    function renderJSONArray(data) {
+        const $rows = data.map((value, index) => {
+            return $('<tr>').append(
+                $('<th>').css('color', 'rgba(150, 150, 150, 1)').text(String(index)),
+                $('<td>').addClass('fa fa-arrow-right'),
+                $('<td>').html(renderJSON(value))
+            );
+        });
+        return $('<table>').addClass('table table-striped').html($('<tbody>').append($rows));
+    }
+
+    function renderJSONObject(data) {
+        const $rows = Object.entries(data).map(([key, value]) => {
+            return $('<tr>').append(
+                $('<th>').css('color', 'rgba(150, 150, 150, 1)').text(key),
+                $('<td>').addClass('fa fa-arrow-right'),
+                $('<td>').html(renderJSON(value))
+            );
+        });
+        return $('<table>').addClass('table table-striped').html($('<tbody>').append($rows));
+    }
+
+    function renderJSON(data) {
+        switch (typeof data) {
+            case 'string':
+                return $textIn(data);
+            case 'number':
+                return $textIn(String(data));
+            case 'boolean':
+                return $textIn(String(data));
+            case 'object':
+                if (data === null) {
+                    return $textIn('NULL');
+                }
+                if (data instanceof Array) {
+                    return renderJSONArray(data);
+                } else {
+                    return renderJSONObject(data);
+                }
+            default:
+                return $textIn('Not representable: ' + typeof data);
+        }
+    }
+
+    /**
+     * Given a string, returns a jquery object wrapping an html tag which contains the
+     * string set, safely, via text().
+     *
+     * @param {string} text A string to render inside the given tag
+     * @param {string} tag The html tag name used to wrap the provided text; defaults to 'div'
+     * @returns JQuery object
+     */
+    function $textIn(text, tag = 'div') {
```

Comment:
This seems to duplicate the `RenderIn` module from your PR #2385. I haven't gone in detail yet - does this get factored out of here, into there?

Reply:
If it doesn't yet, that's fine, just noting here for future work. I don't want to see this one get bigger...

Reply:
I think there are multiple implementations of testing utilities, due to overlapping PRs and the need for such utilities.
They should shake out over time.